tools/targets/PSOC6.py

```diff
@@ -0,0 +1,167 @@
+            message_func("Aligning start from 0x%x to 0x%x" % (start, aligned_start))
+            alignments.frombytes(ihex.tobinarray(aligned_start, start - 1), aligned_start)
+        aligned_end = end & ~align_mask
+        if end != aligned_end:
+            aligned_end += align
+            message_func("Aligning end from 0x%x to 0x%x" % (end, aligned_end))
+            alignments.frombytes(ihex.tobinarray(end, aligned_end - 1), end)
+    ihex.merge(alignments)
+
+def merge_images(hexf0, hexf1=None):
+    ihex = IntelHex()
+    ihex.padding = 0x00
+    ihex.loadfile(hexf0, "hex")
+    if  hexf1 is not None:
+        # get chip ID from metadata and compare
+        ihex1 = IntelHex(hexf1)
+        type0 = ihex.tobinarray(start=0x90500002, size=4)
+        type1 = ihex1.tobinarray(start=0x90500002, size=4)
+        if type0 != type1:
+            raise HookError(
+                "Incompatible processor type: %s in '%s' and 0x%s in '%s'"
+                 % (hexf0, type0, hexf1, type1))
+        ihex.merge(ihex1, 'ignore')
+    return ihex
+
+def complete_func(message_func, elf0, hexf0, hexf1=None, dest=None):
+    message_func("Postprocessing %s -> %s" % (elf0, hexf0))
+    ihex = merge_images(hexf0, hexf1)
+    patch(message_func, ihex, hexf0)
+    ihex.write_hex_file(dest if dest else hexf0, write_start_addr=False, byte_count=64)
+
+def check_matching_features(image_features, target_features):
+    for feature in image_features:
+        if not (feature in target_features):
+            return False
+    return True
+
+# Find Cortex M0 boot image proper for the application image.
+def find_cm0_images(toolchain, resources, elf, hexf):
+    # Scan to find the actual paths of m0 image
+    # First check if user-compiled image exists.
+    params, _ = toolchain.config.get_config_data()
+    dual_core_enabled = params['target.sub-target-build-enable'].value
+    if dual_core_enabled:
+        if "BLE" in toolchain.target.features:
+            raise ConfigException("Feature 'BLE' not compatible with dual core configuration.")
+        m0target = toolchain.target.sub_target
+        m4target = toolchain.target.name
+        m0hexf = hexf.replace(m4target, m0target)
+        if not os.path.isfile(m0hexf):
+            raise ConfigException("Matching M0 core hex image not found.")
+    else:
+        # Try default image.
+        m0hexf = None
+        m0_images = toolchain.target.M0_CORE_IMAGE
+        # convert into a list if needed
+        if not isinstance(m0_images, list):
+            m0_images = [m0_images]
+
+        # find an image with matching features
+        try:
+            target_features = toolchain.target.features
+        except AttributeError:
+            target_features = []
+            pass
+
+        for image in m0_images:
+            features = image['features']
+            if check_matching_features(image['features'], target_features):
+                for hexf in resources.hex_files:
```

Comment:
`resources.hex_files` is the names of all of the hex files in the build. What you want is the paths for every hex file in the build. This is available as:
```python
from tools.resources import FileType
resources.get_file_paths(FileType)
```
The current code will fail in the online compiler.